### PR TITLE
add possibility to ignore events from specified processes

### PR DESCRIPTION
**Pull request by disaykin**
Add possibility to exclude some processes from monitoring

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by tklengyel** [comment]
I don't think this feature should be in libdrakvuf, I think the right spot would be plugins_ex and the plugins that need it can opt-in.

**Comment by disaykin** [comment]
I need to exlude all events for some PIDs. Doing it on per-plugin basis is more complex work.

**Review by tklengyel — CHANGES_REQUESTED** [review]

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/drakvuf.cpp [hunk 1]
```diff
@@ -214,11 +214,29 @@ drakvuf_c::drakvuf_c(const char* domain,
...
     uint64_t limited_traps_ttl,
+    const std::set<uint64_t>& ignored_processes_,
     bool libdrakvuf_get_userid)
     : leave_paused{ leave_paused }
     , format{ output }
 {
-    if (!drakvuf_init(&drakvuf, domain, json_kernel_path, json_wow_path, libvmi_conf, kpgd, fast_singlestep, limited_traps_ttl, libdrakvuf_get_userid))
+    GSList* ignored_processes = NULL;
+    for (auto pid : ignored_processes_)
+        ignored_processes = g_slist_prepend(ignored_processes, (gpointer)pid);
```
[the remark this answers is not in the document]
Comment (disaykin) [reply src/drakvuf.cpp]: Thanks

File: src/drakvuf.cpp [hunk 2]
```diff
@@ -214,11 +214,29 @@ drakvuf_c::drakvuf_c(const char* domain,
...
+            json_wow_path,
+            libvmi_conf,
+            kpgd,
+            fast_singlestep,
+            limited_traps_ttl,
+            ignored_processes,
+            libdrakvuf_get_userid
+        );
+
+    if (!success)
```
[the remark this answers is not in the document]
Comment (disaykin) [reply src/drakvuf.cpp:239]: Done. Thanks.

File: src/libdrakvuf/drakvuf.c [hunk 3]
```diff
@@ -1146,3 +1160,9 @@ void drakvuf_intercept_process_add(drakvuf_t drakvuf, char* process_name, vmi_pi
     process->strict = strict;
     drakvuf->context_switch_intercept_processes = g_slist_prepend(drakvuf->context_switch_intercept_processes, process);
 }
+
+bool drakvuf_is_ignored_process(drakvuf_t drakvuf, vmi_pid_t pid)
+{
+    uint64_t key = pid;
+    return g_slist_find(drakvuf->ignored_processes, (gconstpointer)key) != NULL;
```
Comment (tklengyel) [inline src/libdrakvuf/drakvuf.c:1167]: GUINT_TO_POINTER

**Review by tklengyel — CHANGES_REQUESTED** [review]

File: src/drakvuf.cpp [hunk 4]
```diff
@@ -214,13 +214,35 @@ drakvuf_c::drakvuf_c(const char* domain,
...
+    const std::set<uint64_t>& ignored_processes_,
     bool libdrakvuf_get_userid)
     : leave_paused{ leave_paused }
     , format{ output }
 {
-    if (!drakvuf_init(&drakvuf, domain, json_kernel_path, json_wow_path, libvmi_conf, kpgd, fast_singlestep, limited_traps_ttl, libdrakvuf_get_userid))
+    GSList* ignored_processes = NULL;
+    for (auto pid : ignored_processes_)
+    {
+        gconstpointer key = GUINT_TO_POINTER(pid);
```
Comment (tklengyel) [inline src/drakvuf.cpp:225]: No need to make it into a gconstpointer and have a separate variable on the stack.

Comment (disaykin) [reply src/drakvuf.cpp]: OK

**Pull request merged**